Samples/Swift/DaysUntilBirthday/Shared/ViewModels/AuthenticationViewModel.swift

```diff
@@ -47,7 +47,35 @@ final class AuthenticationViewModel: ObservableObject {
 
   /// Signs the user in.
   func signIn() {
```

Comment:
Should this be async?

Reply:
Not as things stand now. The method currently uses a `Task` to update the sign in state on the main actor, which is required for updating observed objects that will impact view state (as this does). 

Also this method is given as the action in the `GoogleSignInButton`'s initializer. We may want to have another initializer for the button that does take an `async` method, but I think this is a separate conversation and project.

Reply:
Sounds reasonable.  Just looking for ways to minimize the `Task` overhead.